kotsu/run.py

```diff
@@ -55,9 +78,11 @@ def run(
                     "runtime_secs": elapsed_secs,
                 }
             )
-            results_list.append(results)
+            results_df = results_df.append(results, ignore_index=True)
```

Comment:
We might consider using the version that appends to a list before doing a concat, as per the guidance in [the documentation for DataFrame.append](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.append.html)

Reply:
Done

and on 
> we might also consider breaking up the run function a bit since its getting quite long and with various ifs etc - eg a private function to add the meta data that does the privileged key check

Done